mern-ecommerce/controllers/user.js

```diff
@@ -40,6 +41,20 @@ exports.update = (req, res) => {
   );
 };
 
+exports.productById = (req, res, next, id) => { 
+  Product.findById(id) 
+    .populate('products.product', 'name price')
```

Comment:
The populate method should be consistent in its usage. It's currently using both camelCase and underscore notation for field names. It should be consistent throughout the codebase. Consider using underscore notation for field names. 

Suggested change: `.populate('products_product', 'name price')`